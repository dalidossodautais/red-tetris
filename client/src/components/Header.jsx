import React from "react";
import AppBar from "@material-ui/core/AppBar";
import { makeStyles } from "@material-ui/core";
import Spacing from "./Spacing";
import MainTitle from "./texts/MainTitle";

function useClasses() {
  return makeStyles((theme) => ({
    appBar: {
      alignItems: "center",
      backgroundColor: theme.primaryBackgroundColor,
      color: theme.primaryTextColor,
      display: "flex",
      height: "8vh",
      minHeight: "48px",
      justifyContent: "center",
      userSelect: "none",
      width: "100%",
    },
  }))();
}

export default function Header({ children, onClick }) {
  const classes = useClasses();
  return (
    <>
      <AppBar className={classes.appBar} onClick={onClick}>
        <MainTitle>{children}</MainTitle>
      </AppBar>
      <Spacing height="10vh" minHeight="60px" width="100%" />
    </>
  );
}
