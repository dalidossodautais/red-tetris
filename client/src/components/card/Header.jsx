import React from "react";
import styled from "styled-components";
import AppBar from "@material-ui/core/AppBar";
import { makeStyles } from "@material-ui/core/styles";
import Toolbar from "@material-ui/core/Toolbar";
import Icon from "../Icon";
import TextField from "../inputs/TextField";

export const CategoryContainer = styled.span.attrs((props) => ({
  color: props.theme.primaryTextColor,
}))`
  align-items: center;
  color: ${(props) => props.color};
  display: flex;
  justify-content: space-between;
  user-select: none;
`;

const Title = styled.h2``;

function useClasses() {
  return makeStyles((theme) => ({
    appBar: {
      backgroundColor: theme.primaryBackgroundColor,
    },
    toolbar: {
      alignItems: "center",
      display: "flex",
      height: "80px",
      justifyContent: "space-between",
    },
  }))();
}

export default function CardHeader({
  iconName,
  label,
  onChange,
  title,
  value,
  width = "180px",
}) {
  const classes = useClasses();
  return (
    <AppBar className={classes.appBar} position="static">
      <Toolbar className={classes.toolbar}>
        <CategoryContainer>
          <Icon className={classes.icon} marginRight="15px" type={iconName} />
          <Title>{title}</Title>
        </CategoryContainer>
        <TextField
          color="secondary"
          hide={!onChange}
          label={label}
          onChange={onChange}
          value={value}
          variant="outlined"
          width={width}
        />
      </Toolbar>
    </AppBar>
  );
}
