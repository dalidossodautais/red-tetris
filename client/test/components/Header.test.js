import React from "react";
import renderer from "react-test-renderer";
import Header from "../../src/components/Header";

describe("Header", () => {
  test("Standard", () => {
    const component = renderer.create(<Header />);
    // rendering
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
