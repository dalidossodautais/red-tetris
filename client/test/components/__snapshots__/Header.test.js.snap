// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Standard 1`] = `
Array [
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-1 mui-fixed MuiPaper-elevation4"
  >
    <h1
      className="sc-gsTCUz gecxEx"
    />
  </header>,
  <div
    className="sc-bdfBwQ jRZqUG"
    height="10vh"
    width="100%"
  />,
]
`;

exports[`Header Standard 2`] = `
Array [
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-1 mui-fixed MuiPaper-elevation4"
  >
    <h1
      className="sc-gsTCUz gecxEx"
    />
  </header>,
  <div
    className="sc-bdfBwQ jRZqUG"
    height="10vh"
    width="100%"
  />,
]
`;
